### ISSUE TITLE
Upgrade to Linux Mint 20.2 with cinnamon 5.0.4 and Radio++ installed, only right mouse click works

### ISSUE THREAD

**Issue by jongough** [linuxmint/cinnamon-spices-applets#3839]
radio@driglu4it

```
 * Applet version/Build date (to be found in "About.." or in "metadata.json")
 Updated on 2021-07-06 07:08:10

 * Cinnamon version (cinnamon --version)
 5.0.4

 * Distribution - (Mint 17.2, Arch, Fedora 25, etc...)
 Linux Mint 20.2

 * Graphics hardware *and* driver used
Advanced Micro Devices, Inc. [AMD/ATI] Oland PRO [Radeon R7 240/340] (prog-if 00 [VGA controller])

 * Snippet or attachment of ~/.cinnamon/glass.log
[glass.log](https://github.com/linuxmint/cinnamon-spices-applets/files/6794609/glass.log)

 * Snippet or attachment of ~/.xsession-errors
js-Message: 13:36:44.549: JS LOG: [LookingGlass/info] Installing settings for radio@driglu4it
Cjs-Message: 13:36:44.549: JS LOG: [LookingGlass/info] Settings successfully installed for radio@driglu4it

(cinnamon:2334): Clutter-CRITICAL **: 13:36:44.550: clutter_actor_add_child: assertion 'child->priv->parent == NULL' failed
Cjs-Message: 13:36:44.593: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: mintreport (:1.112/org/x/StatusIcon/Icon)
Cjs-Message: 13:36:44.596: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: mintUpdate.py (:1.69/org/x/StatusIcon/Icon)radio
Cjs-Message: 13:36:44.598: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: variety (:1.58/org/x/StatusIcon/Icon_2)
Cjs-Message: 13:36:44.599: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: zulip (:1.58/org/x/StatusIcon/Icon)
Cjs-Message: 13:36:44.601: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: rainlendar2 (:1.58/org/x/StatusIcon/Icon_1)
Cjs-Message: 13:36:45.694: JS LOG: [LookingGlass/info] [weather@mockturtl#121]: Weather Information refreshed
Cjs-Message: 13:37:10.130: JS LOG: [LookingGlass/info] Reloading applet: radio@driglu4it/145

(cinnamon:2334): Clutter-CRITICAL **: 13:37:10.160: clutter_actor_add_child: assertion 'child->priv->parent == NULL' failed
Cjs-Message: 13:37:10.174: JS LOG: [LookingGlass/info] Loaded applet radio@driglu4it in 37 ms

<!-- Author list: https://github.com/linuxmint/cinnamon-spices-applets/blob/master/.github/CODEOWNERS -->

**Notify author of applet**
@

**Issue**
After upgrading to Cinnamon 5.0.4 from 4.8.2 on Linux Mint Radio++ will not respond to left mouse click in the applet area. Right click brings up the 'About/Configure/Remove' menu as expected.


**Steps to reproduce**
Install latest Radio++ on Linux Mint 20.1 (works OK, left click responds)
Upgrade to Linux Mint 20.2 with Cinnamon 5.0.4
Try using Radio++
* Right click brings up About/Configure/Remove menu
* Left click appears to do nothing

**Expected behaviour**
On left mouse click would expect to see list of stations that can be started. 
Having seen this version on another Linux Mint install I know that the left mouse click popup is bigger than before with more items on it.


**Other information**
None
```

**Comment by jonath92** [comment]
Thanks for reporting the issue @jongough  and sorry for the inconvenience. This is not a general problem with Linux Mint 20.2 (i.e. cinnamon 5.0.4) as it works without problems on my pc which also has this version installed. I also have tested it a couple of day ago on Linux Mint 20.2 booted from an USB stick. 

> After upgrading to Cinnamon 5.0.4 from 4.8.2 on Linux Mint Radio++ will not respond to left mouse click in the applet area. [...] On left mouse click would expect to see list of stations that can be started.  

The radio applets uses [mpv](https://mpv.io) as media player and [this plugin](https://github.com/hoyon/mpv-mpris) to allow remote control of mpv. If mpv is not installed or the plugin not placed in the applets config folder `~/.cinnamon/configs/radio@driglu4it`, it is not shown the popup menu when clicking on the applet but this download Confirmation Dialog instead: 

![image](https://user-images.githubusercontent.com/49979415/125189346-f10a6100-e237-11eb-932c-2040f5c3831b.png)

When you click on `Download plugin at my own Risk`, it should be opened the following dialog after a couple of seconds: 

![image](https://user-images.githubusercontent.com/49979415/125189411-3d55a100-e238-11eb-95ab-c872dd515bb1.png)

And after both are installed, the popup menu is shown. 
  
When you click on the applet, the "Download Plugin dialog" is shown in front of all other windows, but I think it still can be overseen and when you than click on another window, the other window is pushed to the front. Clicking on the applet again, however doesn't bring the "Download plugin dialog" in front again. Also there isn't shown an icon in the window list. I think this is something I can improve. But In the meantime please try again to click on the applet and test if you can find the "Mpris download dialog" (maybe behind other windows). In case this doesn't work, you still can download the plugin and install mpv with the following commands: 

```
wget https://github.com/hoyon/mpv-mpris/releases/download/0.5/mpris.so -P ~/.cinnamon/configs/radio@driglu4it
sudo apt install mpv
``` 

Please let me know if this fixed the issue.

**Comment by jongough** [comment]
I have uninstalled and installed mpv, I have downloaded the mpris.so module and placed it in ~/.cinnamon/configs/radio@driglu4it directory and marked it executable. I have uninstalled and reinstalled radio++ but still not luck. I can get the right click window, but the left click appears to do nothing.

I have just done another uninstall/remove, download/install and looking in the ~/.xsession-errors file I can see:
```
Cjs-Message: 07:20:22.086: JS LOG: [LookingGlass/info] Reloading applet: systray@cinnamon.org/85
Cjs-Message: 07:20:22.087: JS LOG: [LookingGlass/info] Reloading applet: keyboard@cinnamon.org/86
Cjs-Message: 07:20:22.088: JS LOG: [LookingGlass/info] Reloading applet: removable-drives@cinnamon.org/88
Cjs-Message: 07:20:22.089: JS LOG: [LookingGlass/info] Reloading applet: calendar@cinnamon.org/92
Cjs-Message: 07:20:22.090: JS LOG: [LookingGlass/info] Reloading applet: notifications@cinnamon.org/119
Cjs-Message: 07:20:22.091: JS LOG: [LookingGlass/info] Reloading applet: xapp-status@cinnamon.org/120
Cjs-Message: 07:20:22.091: JS LOG: [LookingGlass/info] Reloading applet: weather@mockturtl/121
Cjs-Message: 07:20:22.092: JS LOG: [LookingGlass/info] [weather@mockturtl#121]: Removing applet instance...
Cjs-Message: 07:20:22.092: JS LOG: [LookingGlass/info] Reloading applet: multicore-sys-monitor@ccadeptic23/131
Cjs-Message: 07:20:22.093: JS LOG: [LookingGlass/info] Reloading applet: SpicesUpdate@claudiux/134
Cjs-Message: 07:20:22.101: JS LOG: [LookingGlass/info] Adding systray: explorer.exe (24x24px)
Cjs-Message: 07:20:22.101: JS LOG: [LookingGlass/info] Resized explorer.exe with normalized size (24x24px)
Cjs-Message: 07:20:22.108: JS LOG: [LookingGlass/info] Adding systray: explorer.exe (24x24px)
Cjs-Message: 07:20:22.109: JS LOG: [LookingGlass/info] Resized explorer.exe with normalized size (24x24px)
Cjs-Message: 07:20:22.293: JS LOG: [LookingGlass/info] Installing settings for radio@driglu4it
Cjs-Message: 07:20:22.294: JS LOG: [LookingGlass/info] Settings successfully installed for radio@driglu4it

(cinnamon:2569): Clutter-CRITICAL **: 07:20:22.295: clutter_actor_add_child: assertion 'child->priv->parent == NULL' failed
Cjs-Message: 07:20:22.332: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: mintUpdate.py (:1.66/org/x/StatusIcon/Icon)
Cjs-Message: 07:20:22.339: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: mintreport (:1.111/org/x/StatusIcon/Icon)
Cjs-Message: 07:20:22.342: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: virt-manager (:1.52/org/x/StatusIcon/Icon_4)
Cjs-Message: 07:20:22.344: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: variety (:1.52/org/x/StatusIcon/Icon_2)
Cjs-Message: 07:20:22.348: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: rainlendar2 (:1.52/org/x/StatusIcon/Icon)
Cjs-Message: 07:20:22.351: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: qbittorrent (:1.52/org/x/StatusIcon/Icon_3)
Cjs-Message: 07:20:22.354: JS LOG: [LookingGlass/info] Adding XAppStatusIcon: zulip (:1.52/org/x/StatusIcon/Icon_1)
Cjs-Message: 07:20:23.446: JS LOG: [LookingGlass/info] [weather@mockturtl#121]: Weather Information refreshed
```
I have just tried the two commands (wget & apt) and it has made no difference.
```
sudo apt install mpv
[sudo] password for jon:    
Reading package lists... Done
Building dependency tree       
Reading state information... Done
mpv is already the newest version (0.32.0-1ubuntu1).
0 to upgrade, 0 to newly install, 0 to remove and 0 not to upgrade.
```
I have also noticed that all other apps when you roll over them have a little popup box giving information. Radio++ does not do that even though the applet icon is underlined (on the failing machine all the applets are underlined when passing the mouse over them, on other systems that work the whole applet is highlighted).

Is there anyway to get more debugging information from the plugin? Or are there any other runtime logs I can look at to see what is happening? I know little about the cinnamon applets and nothing about how to debug them. However, I do build and support open source applications on Linux, so have a fairly comprehensive setup for testing.

**Comment by jongough** [comment]
I have found what was blocking the running of the applet. I noticed in ~/.loca/share/cinnamon/applets/readio@driglu4it there were both a '3.0' and a '4.6' directory. It looks like '3.0' is the old one that is not currently used so I renamed it '3.0-old', deactivated and then activated the radio and it worked. So it would appear that versions that use 4.6 should remove 3.0 when they are installed. Not sure how to do this, but....

I have done an install on a new Linux Mint 20.0 virtual machine, then upgraded it to 20.2 (to simulate my main machine) and it all works OK. So I suspect there is something in Mint/Cinnamon that is keeping knowledge of 3.0 and is using this by default on my main machine. The virtual machine has the same radio directory/file structure in the applets directory but must be using the '4.6' directory.

Hope this helps.

**Comment by jonath92** [comment]
ok Thanks @jongough. I am happy you fixed it and thanks for the info. As this is as far as I can see a problem of cinnamon and not the applet, I think there is nothing I can do to fix it.

**Comment by jongough** [comment]
It would appear to be a problem with upgrading. I believe I may have been using 3.0 and the system was upgraded to 4.6. With the latest upgrade to Linux Mint 20.2 the 3.0 version non-longer worked and 4.6 was needed. When I renamed the 3.0 directory the system then resolved to use the 4.6 version. I do not know the mechanism as I have no real knowledge of cinnamon nor the plugin applets.

Anyway, as 4.6 is the current version when an upgrade is done the 3.0 directory should be deleted and when doing a new install the 3.0 directory should not be created. This I believe you can do with the packaging and the files that are available from git.

**Comment by jonath92** [comment]
> Anyway, as 4.6 is the current version when an upgrade is done the 3.0 directory should be deleted and when doing a new install the 3.0 directory should not be created

Yes, I agree with you. 

> This I believe you can do with the packaging and the files that are available from git.

Unfortunately I can't. In order to publish two different versions of the applet, which is necessary as the mpris plugin doesn't work on Linux Mint 19 and lower, I have to create a `3.0` and a `4.6` directory inside the `files` directory on Git. I have no impact how the files are than packed and distributed. This is managed by the cinnamon team. 

To be honest, I still don't really understand what happens to you because even if for some weird reason the 3.0 version has been loaded and not the 4.6 version, there should be opened a download dialog and a notification or the popup menu itself. Also what I don't understand is, you said that you have been upgrading from 20.0 to 20.2 but in that case you already should have been used the 4.6 version before and not the 3.0 version. 

Long story short, I understand that this behavior has been frustrating for you and I appreciate that you reported the issue as this might help other users in future. But from my point of view, this seems to be a not reproducible corner case which most likely has been happened due to something weird going on in cinnamon and not in the applet itself and therefore I don't see anything reasonable I can do to fix it.

**Comment by jongough** [comment]
I have been using Radio++ for quite some time (cant remember exactly how long), but I may have been on Mint 19 or 18 when I installed this spice. I did notice someone else in the forum had a similar issue, so the renaming of the 3.0 directory to something else to get it all working and then renaming it back may be the simple solution.

I am happy that it is now working again as I use it most days.
